base/hosts.c

```diff
@@ -1535,6 +1535,60 @@ gvm_hosts_exclude (gvm_hosts_t *hosts, const char *excluded_str)
   return gvm_hosts_exclude_with_max (hosts, excluded_str, 0);
 }
 
+/**
+ * @brief Creates a new gvm_host_t from the provided host_str. host_str can
+ * consist of hostname, IPv4 address or IPv6 address.
+ *
+ * @param[in] host_str The host string. A copy will be created for the returned
+ * struct.
+ *
+ * @return NULL if error. Otherwise, a single host structure that should be put
+ * into a gvm_hosts_t structure for freeing with @ref gvm_hosts_free or
+ * freed directly via @ref gvm_host_free.
+ */
+gvm_host_t *
+gvm_host_from_str (const gchar *host_str)
+{
+  if (host_str == NULL)
+    {
+      return NULL;
+    }
+  int host_type;
```

Comment:
I would really like to see this for variables:
 - always declared at the top of a block,
 - never initialised in the declaration, but in a new statement separated by a newline from the declarations,
 - comments about the block always come after the declarations (but before the initialisations).

Makes the code more predictable.